src/components/pages/Publish/FormAlgoPublish.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactElement, useEffect, FormEvent, ChangeEvent } from 'react'
+import styles from './FormPublish.module.css'
+import { useOcean } from '@oceanprotocol/react'
+import { useFormikContext, Field, Form, FormikContextType } from 'formik'
+import Input from '../../atoms/Input'
+import Button from '../../atoms/Button'
+import { FormContent, FormFieldProps } from '../../../@types/Form'
+import { AlgorithmPublishForm } from '../../../@types/MetaData'
+
+export default function FormPublish({
+  content
+}: {
+  content: FormContent
+}): ReactElement {
+  const { ocean, account } = useOcean()
+  const {
+    status,
+    setStatus,
+    isValid,
+    setErrors,
+    setTouched,
+    resetForm,
+    initialValues,
+    validateField,
+    setFieldValue
+  }: FormikContextType<AlgorithmPublishForm> = useFormikContext()
+
+  // reset form validation on every mount
+  useEffect(() => {
+    setErrors({})
+    setTouched({})
+    resetForm({ values: initialValues, status: 'empty' })
```

Comment:
this might actually lead to problems now with 2 forms behind tabs cause it will clear all entered (and saved in `localStorage`) form values when user switches tabs. And this saving form data and restore on mount is broken in this PR. We haven't fully defined yet how this feature works UX-wise with 2 forms but first guess would be that both forms save their state, and only successfully submitting one of them clears their `localStorage`

Reply:
why don't we have 2 separate keys for the localState `ocean-publish-form-data` and `ocean-publish-form-algo`

Reply:
right, that might be the reason it breaks right now, would be great if it's as simple as that to restore functionality